### ISSUE TITLE
Svelte language server out of heap memory

### ISSUE THREAD

**Issue by WoodyWoodsta** [zed-industries/zed#12546]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

After a short while, the svelte-language-server crashes with "FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory".

I'm not sure if this is an issue with the language server itself because I've not come across this issue before in vscode.

### Environment

Zed: v0.137.6 (Zed)
OS: macOS 14.5.0
Memory: 32 GiB
Architecture: x86_64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

```
stderr: SnapshotManager File Statistics:
stderr: Project files: 565
stderr: Svelte files: 242
stderr: From node_modules: 1258
stderr: Total: 1823
stderr:
stderr: <--- Last few GCs --->
stderr:
stderr: [33211:0x7fad88040000]   279301 ms: Mark-sweep 4050.4 (4142.2) -> 4050.0 (4142.0) MB, 1426.6 / 0.0 ms  (average mu = 0.580, current mu = 0.169) allocation failure; scavenge might not succeed
stderr: [33211:0x7fad88040000]   282145 ms: Mark-sweep 4065.8 (4142.0) -> 4065.4 (4173.2) MB, 2793.8 / 0.0 ms  (average mu = 0.308, current mu = 0.018) allocation failure; scavenge might not succeed
stderr:
stderr:
stderr: <--- JS stacktrace --->
stderr:
stderr: FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
stderr: 1: 0x102e74fb5 node::Abort() (.cold.1) [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
stderr: 2: 0x1018f5a29 node::Abort() [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
stderr: 3: 0x1018f5c0e node::OOMErrorHandler(char const*, bool) [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
stderr: 4: 0x101a84cc3 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
stderr: 5: 0x101c4d975 v8::internal::Heap::FatalProcessOutOfMemory(char const*) [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
stderr: 6: 0x101c4c352 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
stderr: 7: 0x101c3e68a v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
stderr: 8: 0x101c3f005 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
stderr: 9: 0x101c2147e v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
stderr: 10: 0x10205728c v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
stderr: 11: 0x10244cdb9 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_NoBuiltinExit [***/Zed/node/node-v18.15.0-darwin-x64/bin/node]
```

**Comment by notpeter** [comment]
I'm going to go ahead and close this as unactionable because we do not have a way to reproduce this and have not seen others reporting the same thing.

If you have a reproduction test case (e.g. an open source repo which triggers this) I'm happy to reopen, but without that, there's not much we can do to investigate this further. Thanks for reporting.

**Comment by carterharrison** [comment]
If anyone else runs into this issue, I encountered the same thing. I don't think this has anything to do with Zed? I got a similar issue in vscode (I think vscode just ran the language server with access to more memory, however this just led to everything being extremely slow). I found that one of my dependencies had a massive amount of svelte files (an icon library), which led to a ton of memory usage in the language server. Excluding the dependency solved it for me -- not sure if thats the best way to solve it.
